tests/e2e_tests/l1_to_l3_pol_e2e.py

```diff
@@ -0,0 +1,472 @@
+    # (2) Validate Input Images
+    # ================================================================================
+    logger.info('='*80)
+    logger.info('Test Case 1: Input L1 Image Data Format and Content')
+    logger.info('='*80)
+    
+    # Filter to only include L1 files as inputs
+    all_files = os.listdir(l1_datadir)
+    input_files = [f for f in all_files if f.endswith('l1_.fits')]
+    if not input_files:
+        raise FileNotFoundError(f"No files ending in 'l1_.fits' found in {l1_datadir}")
+    
+    input_data_filelist = [os.path.join(l1_datadir, f) for f in input_files]
+    
+    # Create input_data subfolder
+    input_data_dir = os.path.join(l3_outputdir, 'input_l1')
+    if not os.path.exists(input_data_dir):
+        os.makedirs(input_data_dir)
+
+    # Copy files to input_data directory and update file list
+    input_data_filelist = [
+        shutil.copy2(file_path, os.path.join(input_data_dir, os.path.basename(file_path)))
+        for file_path in input_data_filelist
+    ] 
+    
+    # Validate all input images
+    input_dataset = data.Dataset(input_data_filelist)
+    
+    for i, (frame, filepath) in enumerate(zip(input_dataset, input_data_filelist)):
+        frame_info = f"L1 Input Frame {i}"
+        
+        check_filename_convention(os.path.basename(filepath), 'cgi_*_l1_.fits', frame_info, logger, data_level='l1_')
+        verify_header_keywords(frame.ext_hdr, {'DATALVL': 'L1'}, frame_info, logger)
+        
+        
+        # Verify HDU count
+        try:
+            with fits.open(filepath) as hdul:
+                verify_hdu_count(hdul, 4, frame_info, logger) 
+        except Exception as e:
+            logger.info(f"{frame_info}: HDU count verification failed. Error: {str(e)}. FAIL")
+        
+        # Check dimensions
+        logger.info(f"{frame_info}: Data shape {frame.data.shape}")
+        
+        logger.info("")
+    
+    logger.info(f"Total input images validated: {len(input_dataset)}")
+    logger.info('')
+    
+    # ================================================================================
+    # (3) Run Processing Pipeline
+    # ================================================================================
+    logger.info('='*80)
+    logger.info('Running L1 -> L2b -> L3 polarimetry data processing pipeline')
+    logger.info('='*80)
+    
+    # Step 1: L1 -> L2b
+    logger.info('Step 1: Running L1 to L2b recipe...')
+    walker.walk_corgidrp(input_data_filelist, "", l3_outputdir, template="l1_to_l2b.json")
+    
+    l2b_files = [f for f in os.listdir(l3_outputdir) if f.endswith('_l2b.fits')]
+    l2b_filelist = [os.path.join(l3_outputdir, f) for f in l2b_files]
+    logger.info(f'L1 to L2b complete. Generated {len(l2b_filelist)} L2b files.')
+    logger.info('')
+    
+    # Step 2: L2b -> L3 
+    logger.info('Step 2: Running L2b to L3 polarimetry recipe...')
+    walker.walk_corgidrp(l2b_filelist, "", l3_outputdir, template="l2b_to_l3_pol.json")
```

Comment:
if we add auto identification of pol recipe into the walker, we don't need to specify the template explicitly here.